nixos/modules/services/web-apps/gancio.nix

```diff
@@ -0,0 +1,273 @@
+            Default to {option}`services.gancio.enableACME`
+          '';
+          options.enableACME.default = cfg.enableACME;
+          options.enableACME.defaultText = lib.literalMD ''
+            Default to {option}`services.gancio.enableACME`
+          '';
+        }
+      );
+      default = { };
+      example = lib.literalExpression ''
+        {
+          serverAliases = [ "gancio.''${config.networking.domain}" ];
+        }
+      '';
+      description = lib.mdDoc "Extra configuration for the nginx virtual host of gancio.";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    users.users.${cfg.user} = {
+      isSystemUser = true;
+      group = cfg.user;
+      home = "/var/lib/${cfg.user}";
+    };
+    users.groups.${cfg.user} = { };
+
+    systemd.services.gancio =
+      let
+        configFile = json.generate "gancio-config.json" (
+          lib.recursiveUpdate cfg.extraSettings {
+            baseurl = "http${
+              lib.optionalString config.services.nginx.virtualHosts."${cfg.hostName}".enableACME "s"
+            }://${cfg.hostName}";
+            server = {
+              host = cfg.listenHost;
+              port = cfg.listenPort;
+            };
+            log_level = cfg.logLevel;
+            log_path = "/var/log/${cfg.user}";
+            db =
+              {
+                dialect = cfg.dbDialect;
+              }
+              // lib.optionalAttrs (cfg.dbDialect == "sqlite") { storage = "/var/lib/${cfg.user}/db.sqlite"; }
+              // lib.optionalAttrs (cfg.dbDialect == "postgres") {
+                host = "/run/postgresql";
+                database = cfg.user;
+              };
+            user_locale = "/var/lib/${cfg.user}/user_locale";
+            upload_path = "/var/lib/${cfg.user}/uploads";
+          }
+        );
+      in
+      {
+        description = "Gancio server";
+        documentation = [ "https://gancio.org/" ];
+
+        wantedBy = [ "multi-user.target" ];
+        after =
+          [ "network.target" ]
+          ++ lib.optional (cfg.dbDialect == "postgres") "postgresql.service";
+
+        environment = {
+          NODE_ENV = "production";
+        };
+
+        preStart = ''
+          ln -sf ${configFile} config.json
+
+          rm -rf user_locale
+          mkdir -p user_locale
+          ${lib.concatStringsSep "\n" (
+            lib.mapAttrsToList
+              (l: c: "ln -sf ${json.generate "gancio-${l}-locale.json" c} user_locale/${l}.json")
+              cfg.userLocale
+          )}
+
+          rm -rf plugins
+          mkdir -p plugins
+          ${lib.concatMapStringsSep "\n" (p: "ln -sf ${p} plugins/") cfg.plugins}
+        '';
+
+        serviceConfig = {
+          ExecStart = "${cfg.package}/bin/gancio start";
+          StateDirectory = cfg.user;
+          WorkingDirectory = "/var/lib/${cfg.user}";
+          LogsDirectory = cfg.user;
```

Comment:
We are not using journalctl?

Reply:
it's also logging to the console/journalctl with additional debug logs, but currently the logs directory is mandatory in production mode.